gladier/managers/flows_manager.py

```diff
@@ -0,0 +1,387 @@
+            'group': 'urn:globus:groups:id:',
+            'identity': 'urn:globus:auth:identity:'
+        }
+        if id_type not in URN_PREFIXES:
+            raise gladier.exc.DevelopmentException('"id_type" must be one of '
+                                                   f'{URN_PREFIXES.keys()}. Got: {id_type}')
+        return f'{URN_PREFIXES[id_type]}{uuid}'
+
+    def get_flow_permission(self, permission_type, identities=None):
+        """
+        This function is a generic shim that should work for most Gladier clients that
+        want basic permissions that will work with a single Globus Group. This method can be
+        overridden to change any of the automate defaults:
+
+        permission_type for deploying flows:
+            'flow_viewers', 'flow_starters', 'flow_administrators',
+
+        permission_type for running flows:
+            'run_managers', 'run_monitors'
+
+        By default, always returns either None for using automate defaults, or setting every
+        permission_type above to use the set client `globus_group`.
+        """
+        if identities is None and self.globus_group:
+            identities = [self.get_globus_urn(self.globus_group)]
+        permission_types = {
+            'flow_viewers', 'flow_starters', 'flow_administrators', 'run_managers', 'run_monitors'
+        }
+        if permission_type not in permission_types:
+            raise gladier.exc.DevelopmentException(f'permission_type must be one of '
+                                                   f'{permission_types}')
+        return identities
+
+    def get_flow_id(self):
+        return self.storage.get_value('flow_id')
+
+    def validate_flow(self, flow_definition: dict):
+        """
+        Get the current flow id for the current Gladier flow definition.
+        If self.auto_register is True, it will automatically (re)register a flow if it
+        has changed on disk, otherwise raising exceptions.
+
+        :param flow_definition: A complete deployable Flow Definition. Will replace existing
+        flow definition if it differs.
+
+        :raises: gladier.exc.FlowObsolete
+        :raises: gladier.exc.NoFlowRegistered
+        """
+        flow_id = self.get_flow_id()
+        flow_scope = self.storage.get_value('flow_scope')
+        flow_checksum = self.storage.get_value('flow_checksum')
+        if not flow_id or not flow_scope:
+            if self.auto_registration is False:
+                raise gladier.exc.NoFlowRegistered(
+                    f'No flow registered for {self.config_filename} under section {self.section}')
+            self.register_flow(flow_definition)
+        elif flow_checksum != self.get_flow_checksum(flow_definition):
+            if self.auto_registration is False:
+                raise gladier.exc.FlowObsolete(
+                    f'"flow_definition" on {self} has changed and needs to be re-registered.')
+            self.register_flow(flow_definition)
+        return flow_id
```

Comment:
I tried to separate "Validation" from "registration" for both flows and FuncX. One past problem I've had is when something is obsolete, but I want to run anyway. "Force" may be a nice option if that comes up more.